lib/pbench/agent/utils.py

```diff
@@ -390,3 +392,32 @@ def collect_local_info(pbench_bin):
         hostdata[arg] = cp.stdout.strip() if cp.stdout is not None else ""
 
     return (version, seqno, sha1, hostdata)
+
+
+class LocalRemoteHost:
+    """Dynamically build a list of all the local IP addresses so that a one of
```

Comment:
"so that one of two" not "a one of two" 😸  Then again, one could actually make *both* queries, if one wanted to, so maybe just "supporting two simple queries"?

I'm also wondering how many times the caller really knows (or should have to care) whether whatever string it's got is a raw ip address vs a hostname. Granted, there's some optimization here when it *is* known to be an ip, but what's the real benefit of the complication? (Given that accidentally passing a hostname to the ip form can return a false negative.)